<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Mirko Signorelli" />


<title>An introduction to the R package neat</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">An introduction to the R package neat</h1>
<h4 class="author">Mirko Signorelli</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<div id="whats-neat" class="section level2">
<h2>What’s neat?</h2>
<p>neat is the R package that implements NEAT, the Network Enrichment Analysis Test which is presented in Signorelli, M., Vinciotti, V., Wit, E. C. (2016). <em>NEAT: an efficient network enrichment analysis test.</em> BMC Bioinformatics, 17:352.</p>
<p>The article is <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6">freely available from the website of BMC Bioinformatics</a>.</p>
</div>
<div id="whats-network-enrichment-analysis" class="section level2">
<h2>What’s “network” enrichment analysis?</h2>
<p>Network enrichment analysis is an extension of traditional gene enrichment analysis (GEA) tests, which are typically used to provide a characterization of a target gene set by relating it to gene sets (such as Gene Ontologies or KEGG pathways) whose function is already known.</p>
<p>A known limitation of GEA tests is that they ignore associations and dependences between genes. The purpose of network enrichment analysis is thus to integrate GEA tests with information on known relations between genes, represented by means of a gene network.</p>
<p>Loosely speaking, we can say that network enrichment analysis incorporates genetic networks, with their information on gene dependences, into gene enrichment tests. Hence, the name “network” enrichment analysis.</p>
</div>
<div id="get-started" class="section level2">
<h2>Get started</h2>
<p>In order to be able to use the package, you need to install it with</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">install.packages</span>(<span class="st">&#39;neat&#39;</span>)</a></code></pre></div>
<p>and, then, to load it with the command</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(<span class="st">&#39;neat&#39;</span>)</a></code></pre></div>
</div>
</div>
<div id="a-first-example" class="section level1">
<h1>A first example</h1>
<p>Let’s first have a quick look at an example of how a network enrichment analysis can be carried out with NEAT.</p>
<p>The analysis will typically consist of three steps: preparation of the data, computation of the test and inspection of the results.</p>
<div id="preparation-of-the-data" class="section level2">
<h2>Preparation of the data</h2>
<p>Let’s start by loading <code>yeast</code>, a list which contains the data that we will need for the analysis:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw">data</span>(yeast) <span class="co"># load the data</span></a>
<a class="sourceLine" id="cb3-2" title="2"><span class="kw">ls</span>(yeast) <span class="co"># display the content of the list</span></a></code></pre></div>
<pre><code>## [1] &quot;esr1&quot;       &quot;esr2&quot;       &quot;goslimproc&quot; &quot;kegg&quot;       &quot;yeastnet&quot;  
## [6] &quot;ynetgenes&quot;</code></pre>
<p>Let’s say that we are interested to know whether a set of differentially expressed genes, <code>yeast$esr2</code>, can be related to some functional gene sets contained in <code>yeast$goslimproc</code>. Let’s focus the attention on two of these processes, namely ‘response to heat’ and ‘response to starvation’.</p>
<p>Before we can proceed with the analysis, we have to create two lists of gene sets, one (which we will call <code>induced_genes</code>) containing the set of differentially expressed genes and the other (called <code>functional_sets</code>) with the functional sets of interest:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">induced_genes =<span class="st"> </span><span class="kw">list</span>(<span class="st">&#39;ESR 2&#39;</span> =<span class="st"> </span>yeast<span class="op">$</span>esr2) <span class="co"># set of differentially expressed genes </span></a>
<a class="sourceLine" id="cb5-2" title="2"><span class="co">#(ESR 2 is the set of induced ESR genes)</span></a>
<a class="sourceLine" id="cb5-3" title="3">functional_sets =<span class="st"> </span>yeast<span class="op">$</span>goslimproc[<span class="kw">c</span>(<span class="dv">72</span>,<span class="dv">75</span>)] <span class="co"># two functional gene sets of interest: </span></a>
<a class="sourceLine" id="cb5-4" title="4"><span class="co">#response to heat and to starvation</span></a></code></pre></div>
<p>Besides these two lists, we will need two further objects:</p>
<ul>
<li><code>yeast$yeastnet</code>, a two-column matrix that contains YeastNet (a network incorporating known functional couplings between yeast genes, see the help page <code>?yeast</code> for more details);</li>
<li><code>yeast$ynetgenes</code>, a vector containing the names of all the genes that are present in the network.</li>
</ul>
</div>
<div id="computation-of-the-test" class="section level2">
<h2>Computation of the test</h2>
<p>The idea behind NEAT is that if two gene sets are related, then in the network we expect them to be connected by a larger (or smaller) number of links than we would expect to observe by chance. Our null hypothesis, thus, is that if A and B are unrelated, then links are randomly placed between the two groups, so that the total number of links between A and B can be assumed to follow an hypergeometric distribution.</p>
<p>If, however, the number of links that we actually observe between A and B turns out to be significantly different from what we would expect to get if links were placed randomly, then we take this fact as potential evidence of a relation between the two groups and we say that there is ``enrichment&quot; between them.</p>
<p>The computation of the test can be done with the function <code>neat</code> as follows:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1">test =<span class="st"> </span><span class="kw">neat</span>(<span class="dt">alist =</span> induced_genes, <span class="dt">blist =</span> functional_sets, <span class="dt">network =</span> yeast<span class="op">$</span>yeastnet,</a>
<a class="sourceLine" id="cb6-2" title="2">            <span class="dt">nettype =</span> <span class="st">&#39;undirected&#39;</span>, <span class="dt">nodes =</span> yeast<span class="op">$</span>ynetgenes, <span class="dt">alpha =</span> <span class="fl">0.01</span>, <span class="dt">mtc.type =</span> <span class="st">&#39;fdr&#39;</span>)</a></code></pre></div>
</div>
<div id="analysis-of-the-results" class="section level2">
<h2>Analysis of the results</h2>
<p>The results are now saved in the object <code>test</code>, which we can display with the command <code>print</code>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"><span class="kw">print</span>(test)</a></code></pre></div>
<pre><code>##       A                      B nab expected_nab pvalue adjusted_p
## 1 ESR 2       response_to_heat  86         96.9 0.2518     0.2518
## 2 ESR 2 response_to_starvation 459        331.4 0.0000     0.0000
##       conclusion
## 1  No enrichment
## 2 Overenrichment</code></pre>
<p>From the table we can see that the set of differentially expressed genes (<code>ESR 2</code>) is not enriched with respect to the set of genes involved in response to heating, whereas it is overenriched with respect to the set of genes that are responsible for response to starvation (that is to say, the observed number of links, 459, is significantly higher than what we would expect to get by chance, i.e. 331). Thus, we can conclude that genes in <code>ESR 2</code> are regulated when the yeast cell is exposed to starvation, but not when exposed to heating.</p>
</div>
</div>
<div id="a-closer-look-to-the-package" class="section level1">
<h1>A closer look to the package</h1>
<p>The core of the package is the function <code>neat</code>:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1"><span class="kw">neat</span>(alist, blist, network, nettype, nodes, </a>
<a class="sourceLine" id="cb9-2" title="2">     <span class="dt">alpha =</span> <span class="ot">NULL</span>, <span class="dt">mtc.type =</span> <span class="st">&#39;fdr&#39;</span>,</a>
<a class="sourceLine" id="cb9-3" title="3">     <span class="dt">anames =</span> <span class="ot">NULL</span>, <span class="dt">bnames =</span> <span class="ot">NULL</span>)</a></code></pre></div>
<p>The fundamental arguments of the function are:</p>
<ul>
<li><p><code>alist</code> and <code>blist</code>, two lists of gene sets;</p></li>
<li><p><code>network</code>, which can be specified in three different formats;</p></li>
<li><p><code>nettype</code>, either <code>&#39;undirected&#39;</code> or <code>&#39;directed&#39;</code>;</p></li>
<li><p><code>nodes</code>, a vector containing the names of all nodes in the network.</p></li>
</ul>
<p>Moreover, three optional arguments are <code>alpha</code>, which allows to specify the significance level of the test, and <code>anames</code> and <code>bnames</code> (they can be used to name the elements of <code>alist</code> and <code>blist</code>, if not already named). Starting from package version 1.2.0, a further argument <code>mtc.type</code> has been added to allow computation of multiple testing corrections within neat.</p>
<p>As a (toy) example, let’s consider a partially directed network with 7 nodes defined by the following adjacency matrix</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">A =<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow=</span><span class="dv">7</span>, <span class="dt">ncol=</span><span class="dv">7</span>)</a>
<a class="sourceLine" id="cb10-2" title="2">labels =<span class="st"> </span>letters[<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</a>
<a class="sourceLine" id="cb10-3" title="3"><span class="kw">rownames</span>(A) =<span class="st"> </span>labels; <span class="kw">colnames</span>(A) =<span class="st"> </span>labels</a>
<a class="sourceLine" id="cb10-4" title="4">A[<span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>)]=<span class="dv">1</span>; A[<span class="dv">2</span>,<span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">7</span>)]=<span class="dv">1</span>;A[<span class="dv">3</span>,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)]=<span class="dv">1</span>;A[<span class="dv">4</span>,<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">7</span>)]=<span class="dv">1</span>;A[<span class="dv">6</span>,<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">5</span>)]=<span class="dv">1</span>;A[<span class="dv">7</span>,<span class="dv">4</span>]=<span class="dv">1</span></a>
<a class="sourceLine" id="cb10-5" title="5"><span class="kw">print</span>(A)</a></code></pre></div>
<pre><code>##   a b c d e f g
## a 0 1 1 0 0 0 0
## b 0 0 0 0 1 0 1
## c 1 0 0 1 0 0 0
## d 0 1 0 0 1 0 1
## e 0 0 0 0 0 0 0
## f 0 1 0 0 1 0 0
## g 0 0 0 1 0 0 0</code></pre>
<div id="how-to-specify-the-lists-of-gene-sets" class="section level3">
<h3>How to specify the lists of gene sets</h3>
<p>Let’s consider three sets of genes {a,e}, {c,g} and {d,f} and suppose we want to test whether there is enrichment from the first two sets to the third one.</p>
<p>First of all, let’s create a vector for each of the three sets:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">set1 =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;a&#39;</span>,<span class="st">&#39;e&#39;</span>)</a>
<a class="sourceLine" id="cb12-2" title="2">set2 =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;c&#39;</span>,<span class="st">&#39;g&#39;</span>)</a>
<a class="sourceLine" id="cb12-3" title="3">set3 =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;d&#39;</span>,<span class="st">&#39;f&#39;</span>)</a></code></pre></div>
<p>As we want to know whether there is enrichment from <code>set1</code> and <code>set2</code> to <code>set3</code>, we can create two gene lists, one (<code>alist</code>) containing <code>set1</code> and <code>set2</code> and the other (<code>blist</code>) containing <code>set3</code>:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">alist =<span class="st"> </span><span class="kw">list</span>(<span class="st">&#39;set 1&#39;</span> =<span class="st"> </span>set1, <span class="st">&#39;set 2&#39;</span> =<span class="st"> </span>set2)</a>
<a class="sourceLine" id="cb13-2" title="2">blist =<span class="st"> </span><span class="kw">list</span>(<span class="st">&#39;set 3&#39;</span> =<span class="st"> </span>set3)</a></code></pre></div>
</div>
<div id="alternative-network-formats" class="section level3">
<h3>Alternative network formats</h3>
<p>Above we have defined the network with its adjacency matrix <code>A</code>. However, the network can be passed to <code>neat</code> in three alternative formats:</p>
<ul>
<li>a sparse adjacency matrix, e.g.</li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="kw">library</span>(Matrix)</a>
<a class="sourceLine" id="cb14-2" title="2"><span class="kw">as</span>(A, <span class="st">&#39;sparseMatrix&#39;</span>)</a></code></pre></div>
<pre><code>## 7 x 7 sparse Matrix of class &quot;dgCMatrix&quot;
##   a b c d e f g
## a . 1 1 . . . .
## b . . . . 1 . 1
## c 1 . . 1 . . .
## d . 1 . . 1 . 1
## e . . . . . . .
## f . 1 . . 1 . .
## g . . . 1 . . .</code></pre>
<ul>
<li><p>an <code>igraph</code> graph;</p></li>
<li><p>a two-column matrix where every row represents and edge (for directed and mixed networks, parent nodes must be in the first column, and child nodes in the second), e.g.:</p></li>
</ul>
<pre><code>##       [,1] [,2]
##  [1,] &quot;a&quot;  &quot;b&quot; 
##  [2,] &quot;a&quot;  &quot;c&quot; 
##  [3,] &quot;b&quot;  &quot;e&quot; 
##  [4,] &quot;b&quot;  &quot;g&quot; 
##  [5,] &quot;c&quot;  &quot;a&quot; 
##  [6,] &quot;c&quot;  &quot;d&quot; 
##  [7,] &quot;d&quot;  &quot;b&quot; 
##  [8,] &quot;d&quot;  &quot;e&quot; 
##  [9,] &quot;d&quot;  &quot;g&quot; 
## [10,] &quot;f&quot;  &quot;b&quot; 
## [11,] &quot;f&quot;  &quot;e&quot; 
## [12,] &quot;g&quot;  &quot;d&quot;</code></pre>
</div>
<div id="network-type" class="section level3">
<h3>Network type</h3>
<p>Set the argument <code>nettype</code> equal to <code>&#39;undirected&#39;</code> if an undirected network is at hand, and equal to <code>&#39;directed&#39;</code> if you are considering a directed or partially directed network.</p>
</div>
<div id="compute-the-test" class="section level3">
<h3>Compute the test</h3>
<p>Once you have prepared the lists of gene sets and the network, what you need is to run <code>neat</code>, without forgetting to specify the correct <code>nettype</code> (here <code>nettype = &#39;directed&#39;</code>) and the labels of nodes (here <code>nodes = labels</code>):</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">test1 =<span class="st"> </span><span class="kw">neat</span>(<span class="dt">alist =</span> alist, <span class="dt">blist =</span> blist, <span class="dt">network =</span> A, </a>
<a class="sourceLine" id="cb17-2" title="2">             <span class="dt">nettype =</span> <span class="st">&#39;directed&#39;</span>, <span class="dt">nodes =</span> labels)</a>
<a class="sourceLine" id="cb17-3" title="3"><span class="kw">print</span>(test1)</a></code></pre></div>
<pre><code>##       A     B nab expected_nab     pvalue adjusted_p
## 1 set 1 set 3   0       0.3333 0.68181818 0.68181818
## 2 set 2 set 3   2       0.5000 0.04545455 0.09090909</code></pre>
<p>If you want to add to the results a column specifying the conclusion of the test (overenrichment, no enrichment or underenrichment) for a given significance level, you use the option <code>alpha</code>. Note that the conclusion will obviously depend on the type of multiple correction you choose (compare test2 and test3 below):</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1"><span class="co"># results using FDR correction (default since version 1.2.0)</span></a>
<a class="sourceLine" id="cb19-2" title="2">test2 =<span class="st"> </span><span class="kw">neat</span>(<span class="dt">alist =</span> alist, <span class="dt">blist =</span> blist, <span class="dt">network =</span> A, </a>
<a class="sourceLine" id="cb19-3" title="3">             <span class="dt">nettype =</span> <span class="st">&#39;directed&#39;</span>, <span class="dt">nodes =</span> labels, <span class="dt">alpha =</span> <span class="fl">0.05</span>, <span class="dt">mtc.type =</span> <span class="st">&#39;fdr&#39;</span>)</a>
<a class="sourceLine" id="cb19-4" title="4"><span class="kw">print</span>(test2)</a></code></pre></div>
<pre><code>##       A     B nab expected_nab     pvalue adjusted_p    conclusion
## 1 set 1 set 3   0       0.3333 0.68181818 0.68181818 No enrichment
## 2 set 2 set 3   2       0.5000 0.04545455 0.09090909 No enrichment</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1"><span class="co"># results without multiple testing correction (mtc.type = &#39;none&#39;)</span></a>
<a class="sourceLine" id="cb21-2" title="2">test3 =<span class="st"> </span><span class="kw">neat</span>(<span class="dt">alist =</span> alist, <span class="dt">blist =</span> blist, <span class="dt">network =</span> A, </a>
<a class="sourceLine" id="cb21-3" title="3">             <span class="dt">nettype =</span> <span class="st">&#39;directed&#39;</span>, <span class="dt">nodes =</span> labels, <span class="dt">alpha =</span> <span class="fl">0.05</span>, <span class="dt">mtc.type =</span> <span class="st">&#39;none&#39;</span>)</a>
<a class="sourceLine" id="cb21-4" title="4"><span class="kw">print</span>(test3)</a></code></pre></div>
<pre><code>##       A     B nab expected_nab     pvalue adjusted_p     conclusion
## 1 set 1 set 3   0       0.3333 0.68181818 0.68181818  No enrichment
## 2 set 2 set 3   2       0.5000 0.04545455 0.04545455 Overenrichment</code></pre>
</div>
</div>
<div id="further-details-and-material" class="section level1">
<h1>Further details and material</h1>
<p>The aim of this vignette is to provide a quick introduction to the computation of NEAT using R. Here I focused my attention on the fundamental aspects that one needs to use the package.</p>
<p>Further details, functions and examples can be found in <a href="https://cran.r-project.org/package=neat">the manual of the package</a>.</p>
<p>The description of the method is available in an article <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6">which you can read here</a>.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
